src/snowflake/cli/plugins/snowpark/models.py

```diff
@@ -30,3 +47,34 @@ class RequirementWithFiles:
 
     requirement: Requirement
     files: List[str]
+
+
+@dataclass
+class RequirementWithFilesAndDeps(RequirementWithFiles):
+    dependencies: List[str]
+
+    def to_requirement_with_files(self):
+        return RequirementWithFiles(self.requirement, self.files)
+
+
+pip_failed_msg = """pip failed with return code {}.
+            If pip is installed correctly, this may mean you`re trying to install a package
+            that isn't compatible with the host architecture -
+            and generally means it has native libraries."""
+second_chance_msg = """Good news! The following package dependencies can be
+                imported directly from Anaconda, and will be excluded from
+                the zip: {}"""
+
+
+def get_package_name(name: str) -> str:
+    if name.startswith("git+"):
+        pattern = r"github\.com\/([^\/]+)\/([^\/.]+)(\.git)?"
+        if match := re.search(pattern, name):
+            return match.group(2)
+        else:
+            return "package"
```

Comment:
Does it mean we will try to download `package`?

Reply:
A bit of a placeholder currently. I'm not quite sure, what we should return here